src/parameters.py

```diff
@@ -1,20 +1,35 @@
 from pydantic import BaseModel, Field
 from typing import Optional, List
 
+#===========================================I/O Related Parameters===========================================#
+class IOParameters(BaseModel):
+    data_tiled_uri: str = Field(description="tiled uri for image data")
+    data_tiled_api_key: str = Field(description="tiled api key for data client")
+    mask_tiled_uri: Optional[str] = Field(default=None, description="tiled uri for masks")
+    mask_tiled_api_key: Optional[str] = Field(default=None, description="tiled api key for masks")
+    seg_tiled_uri: Optional[str] = Field(default=None, description="tiled uri for segmenation results")
+    seg_tiled_api_key: Optional[str] = Field(default=None, description="tiled api key for segmentation results")
+    uid: str = Field(description="uid for the workflow")
+
 #===========================================General Parameters===========================================#
 class TrainingParameters(BaseModel):
     network: str = Field(description="type of dlsia network used")
     num_classes: int = Field(description="number of classes as output channel")
     num_epochs: int = Field(default=10, description="number of epochs")
     optimizer: str = Field(default="Adam", description="optimizer used for training")
     criterion: str = Field(default="CrossEntropyLoss", description="criterion for loss")
-    weights: Optional[str] = Field(default=None, description="weights per class for imbalanced labeling")
+    weights: Optional[List[int]] = Field(default=None, description="weights per class for imbalanced labeling")
```

Comment:
I believe we settled on leaving this as a string in the yaml file, but parsing it to a list of floats (unless PyTorch expects integers).

A reasonable alternative may also be to use weights according to the class distribution and make this a boolean parameter. The class imbalance could be computed during mask exporting and become additional meta data.

Reply:
I thought the example weights I got was in the format of ` [1.0, 1.0, 1.0]`, double checking on this, would this be a string or essentially this is a list in yaml file.

I have set this to a str earlier, and right now this is what I have `weights: Optional[List[float]]` in parameter validation. Will check again tomorrow.